sdk/core/azure-core/test/ut/azure_libcurl_core_main.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file This test assumes the application is already using libcurl.
+ *
+ */
+
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+
+#include <gtest/gtest.h>
+
+#include <curl/curl.h>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/curl_transport.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/io/body_stream.hpp>
+#include <azure/core/response.hpp>
+
+#include <http/curl/curl_connection_pool_private.hpp>
+#include <http/curl/curl_connection_private.hpp>
+#include <http/curl/curl_session_private.hpp>
+
+#include <cstdlib>
+
+namespace Azure { namespace Core { namespace Test {
+  TEST(SdkWithLibcurl, globalCleanUp)
+  {
+    Azure::Core::Http::Request req(
+        Azure::Core::Http::HttpMethod::Get, Azure::Core::Url("https://httpbin.org/get"));
+
+    {
+      // Creating a new connection with default options
+      Azure::Core::Http::CurlTransportOptions options;
+      auto connection
+          = Azure::Core::Http::_detail::CurlConnectionPool::ExtractOrCreateCurlConnection(
+              req, options);
+
+      auto session = std::make_unique<Azure::Core::Http::CurlSession>(
+          req, std::move(connection), options.HttpKeepAlive);
+      session->Perform(Azure::Core::Context::GetApplicationContext());
+      // Reading all the response
+      session->ReadToEnd(Azure::Core::Context::GetApplicationContext());
+    }
+    // Check that after the connection is gone, it is moved back to the pool
+    EXPECT_EQ(Azure::Core::Http::_detail::CurlConnectionPool::ConnectionPoolIndex.size(), 1);
```

Comment:
From https://dev.azure.com/azure-sdk/internal/_build/results?buildId=833900&view=logs&j=d4086158-a6c8-5dff-fca6-c08bb4399a66&t=aeff5a04-d889-5432-f5d0-c5790c13d58f&l=293
> azure_libcurl_core_main.cpp(54): warning C6326: Potential comparison of a constant with another constant. 

How's that two constants? Looks like a false positive warning, should we disable it?

We do so here:
https://github.com/Azure/azure-sdk-for-cpp/blob/0fd687129ee6adf5546044004dbf74d11c27509e/sdk/core/azure-core/test/ut/CMakeLists.txt#L81-L82

@CaseyCarter - why is MSVC giving this warning?

Reply:
Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/2090

Reply:
That's what we get when we go as pedantic as /permissive- /W4 !!  

GoogleTests uses MACROS which are resolved to code, something like EXPECT_TRUE(...)  needs to be translated to some code which might be ending as if (1 == 1)  ..  (my guess) .   That's why we disable that warning on MSVC for tests.

Reply:
With a recent-ish (VS 2017-era, IIRC?) compiler `/diagnostics:caret` might produce some more helpful info to diagnose the issue.